src/lib/converter/factories/comment.ts

```diff
@@ -32,14 +32,7 @@ export function createComment(node: ts.Node): Comment | undefined {
  * @return TRUE if the given node is the topmost module declaration, FALSE otherwise.
  */
 function isTopmostModuleDeclaration(node: ts.ModuleDeclaration): boolean {
-    if (node.nextContainer && node.nextContainer.kind === ts.SyntaxKind.ModuleDeclaration) {
-        let next = <ts.ModuleDeclaration> node.nextContainer;
-        if (node.name.end + 1 === next.name.pos) {
-            return false;
-        }
-    }
-
-    return true;
+    return node.getChildren().some(ts.isModuleBlock);
```

Comment:
Toss `namespace A.B.C {}` into https://ts-ast-viewer.com/ and it's immediately obvious that a module declaration is "topmost" if it has a module block as an immediate child.